src/plugin.ts

```diff
@@ -6,17 +6,27 @@ import { createAdapters } from "./adapters";
 import { userPullRequest, userSelfAssign, userStartStop, userUnassigned } from "./handlers/user-start-stop";
 import { Context, Env, PluginInputs } from "./types";
 import { addCommentToIssue } from "./utils/issue";
+import { createAppAuth } from "@octokit/auth-app";
 
 export async function startStopTask(inputs: PluginInputs, env: Env) {
   const customOctokit = Octokit.plugin(paginateGraphQL);
   const octokit = new customOctokit({ auth: inputs.authToken });
   const supabase = createClient(env.SUPABASE_URL, env.SUPABASE_KEY);
 
+  const jwtOctokit = new Octokit({
```

Comment:
I don't think this is correct. Why would we need two Octokit instances?

Reply:
hmm indeed.. let me check if i can combine them.
To be able to use `listInstallations`, it needs JWT.

Reply:
I cannot use the provided `inputs.authToken`, I don't think it is a JWT token. 

```
{
            "err": {
              "name": "HttpError",
              "status": 401,
              "response": {
                "url": "https://api.github.com/app/installations",
                "status": 401,
                "headers": {
                  "access-control-allow-origin": "*",
                  "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP,
  X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource,
  X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type,
  X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
                  "content-length": "117",
                  "content-security-policy": "default-src 'none'",
                  "content-type": "application/json; charset=utf-8",
                  "date": "Mon, 21 Oct 2024 19:04:16 GMT",
                  "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
                  "server": "github.com",
                  "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
                  "vary": "Accept-Encoding, Accept, X-Requested-With",
                  "x-content-type-options": "nosniff",
                  "x-frame-options": "deny",
                  "x-github-media-type": "github.v3; format=json",
                  "x-github-request-id": "C484:3BDE26:ADC1C3:AFD8B6:6716A5B0",
                  "x-xss-protection": "0"
                },
                "data": {
                  "message": "A JSON web token could not be decoded",
                  "documentation_url": "https://docs.github.com/rest",
                  "status": "401"
                }
              },
              "request": {
                "method": "GET",
                "url": "https://api.github.com/app/installations",
                "headers": {
                  "accept": "application/vnd.github.v3+json",
                  "user-agent": "octokit-rest.js/20.1.1 octokit-core.js/5.2.0 Cloudflare-Workers",
                  "authorization": "token [REDACTED]"
                },
                "request": {}
              }
            },
            "caller": "startStopTask"
          }
          
```
Not sure what the best way to solve this.

Reply:
The octokit is logged in using a Github App token.

Reply:
to list installs we need app private key do we not?

Reply:
yes, we do need JWT to be exact.

Reply:
What we've done in the past is get it passed in from GitHub Actions. 

Get: https://github.com/ubiquity/cloudflare-deploy-action/blob/main/action.yml#L50-L56

Use: https://github.com/ubiquity/cloudflare-deploy-action/blob/main/action.yml#L62

Reply:
fair enough

Reply:
Still feels wrong to me that we need two octokit instances with different auth and private keys. @whilefoo rfc

Reply:
listing installs is for sure the most convenient and effective way to find out the entire partner network

hack 1: use `ubiquityos_app_slug` against the search API for comments and then filter a `Set` of the orgs which it has appeared in
hack 2: install a secondary app to avoid exposing kernel' pk (guaranteed it seems partner's will have to do this anyway)
hack 3: Auth as the kernel with `clientId` but I think that's the same as exposing the PK

Reply:
With option 1, it means third parties won't be able to use the plugin right? 
Is it also an idea to query all issues in `devpool-directory` and built a set of orgs from the issues there?

Reply:
1. partner's will use the same app slug, if you mean forks/development then they'd change to their own app slug.
2. That is also an effective suggestion

Reply:
Do we need bot's token just to list all installations or also installation tokens from each installation?
One option would be an API endpoint on the kernel to list installations, I'd avoid using bot's private key other than in kernel if possible.

Reply:
yes, we only need bot's token to list all installations
Regarding the API endpoint, is it something that is already possible on the kernel side? If so, I think it is better to move it there